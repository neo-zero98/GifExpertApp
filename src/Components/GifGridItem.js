import React from 'react'
import PropTypes  from "prop-types";

const GifGridItem = ({img}) => {
    return (
        <>
            <div className="card">
                <img src={img.url} />
                <p>{img.title}</p>
            </div>
        </>
    )
}

GifGridItem.protoTypes = {
    img: PropTypes.object.isRequired
}

export default GifGridItem
