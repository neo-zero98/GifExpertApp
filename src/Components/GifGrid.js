import React from 'react'
import { useFetchGifs } from '../Hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {loading && <p>Loading</p>}
            <div className="card-grid animate__fadeIn">
                {
                    data.map( img => (
                        <GifGridItem 
                            key={img.id} 
                            img={img}
                        />
                    )
                )}
            </div>
        </>
    )
}

export default GifGrid;
