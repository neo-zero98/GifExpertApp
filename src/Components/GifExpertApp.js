import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import GifGrid from './GifGrid';

export const GifExpertApp = ({defaultCategories=[]}) => {
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h1>GifExpertApp</h1>   
            <AddCategory setCategories={setCategories}/>
            <hr></hr>

            <ol>
                {
                    categories.map((category) => (
                        // return <li key={categorie}>{categorie}</li>
                        <GifGrid 
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    )
}
