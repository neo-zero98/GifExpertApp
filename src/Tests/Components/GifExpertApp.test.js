import { shallow } from "enzyme";
import { GifExpertApp } from "../../Components/GifExpertApp";
import '@testing-library/jest-dom';

describe('Pruebas de <GifExpertApp />', () => {
    
    test('toMatchSnapshot', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['Naruto', 'One Punch Man'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });    
});
