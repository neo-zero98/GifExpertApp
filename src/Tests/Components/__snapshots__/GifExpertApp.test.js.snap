// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="One Punch Man"
      key="One Punch Man"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas de <GifExpertApp /> toMatchSnapshot 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
